tests/shotgun_authentication_tests/test_session_cache.py

```diff
@@ -9,12 +9,36 @@
 # not expressly granted therein are reserved by Shotgun Software Inc.
 
 from __future__ import with_statement
+import os
+
+import unittest2
 
 from tank_test.tank_test_base import *
 
 from tank_vendor.shotgun_authentication import session_cache
 
 
+class SessionCacheLocationTests(unittest2.TestCase):
+    """
+    Separate test for getting cache location since this method is mocked in the TankTestBase.
+    """
+    def test_get_session_cache_location(self):
+        """
+        Makes sure session cache is overidable with SHOTGUN_HOME env var and that env vars
+        and ~ are substituted.
+        """
+        old_sg_home = os.environ.get("SHOTGUN_HOME")
+        try:
+            os.environ["TK_SHOTGUN_HOME_TEST"] = "shotgun_home_test"
```

Comment:
When reading SHOTGUN_HOME from os.environ we run os.path.expanduser and expandvars on the value so that we have a real path. That is why the tests goes through the trouble of setting a test var.